src/rules/completedDocsRule.ts

```diff
@@ -19,100 +19,415 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+export interface IBlockRequirementDescriptor {
+    visibilities?: Visibility[];
+}
+
+export interface IClassRequirementDescriptor {
+    locations?: Location[];
+    privacies?: Privacy[];
+}
+
+export type RequirementDescriptor = IBlockRequirementDescriptor | IClassRequirementDescriptor;
+
+export interface IRequirementDescriptors {
+    [type: string /* DocType */]: RequirementDescriptor;
+}
+
+export type All = typeof Rule.ALL;
+
+export type DocType = All
+    | typeof Rule.ARGUMENT_CLASSES
```

Comment:
I'm using VS Code, TS 2.1.4. So you're saying that `const a: Visibility = "s";` causes a compilation error for you?